drivers/hmis/spec/requests/hmis/update_form_definition_spec.rb

```diff
@@ -0,0 +1,50 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  include_context 'hmis base setup'
+  let!(:access_control) { create_access_control(hmis_user, ds1) }
+
+  before(:each) do
+    hmis_login(user)
+  end
+
+  let(:mutation) do
+    <<~GRAPHQL
+      mutation UpdateFormDefinition($id: ID!, $input: FormDefinitionInput!) {
+        updateFormDefinition(id: $id, input: $input) {
+          formDefinition {
+            id
+          }
+          #{error_fields}
+        }
+      }
+    GRAPHQL
+  end
+
+  it 'should work when the definition needs to be converted' do
+    input = {
+      definition: '{"__typename":"FormDefinitionJson","item":[{"__typename":"FormItem","item":[{"__typename":"FormItem","item":null,"linkId":"4.19.1","type":"DATE","component":null,"prefix":null,"text":"Assessment Date","briefText":null,"readonlyText":null,"helperText":null,"required":false,"warnIfEmpty":false,"hidden":false,"readOnly":false,"repeats":false,"mapping":null,"pickListReference":null,"size":null,"assessmentDate":true,"prefill":false,"bounds":null,"pickListOptions":null,"initial":null,"dataCollectedAbout":null,"disabledDisplay":"HIDDEN","enableBehavior":"ANY","enableWhen":null,"autofillValues":null},{"__typename":"FormItem","item":null,"linkId":"emergency_contact_name","type":"STRING","component":null,"prefix":null,"text":"Emergency Contact Name 1","briefText":null,"readonlyText":null,"helperText":null,"required":false,"warnIfEmpty":false,"hidden":false,"readOnly":false,"repeats":false,"mapping":{"__typename":"FieldMapping","recordType":null,"fieldName":null,"customFieldKey":"emergency_contact_name"},"pickListReference":null,"size":null,"assessmentDate":null,"prefill":false,"bounds":null,"pickListOptions":null,"initial":null,"dataCollectedAbout":null,"disabledDisplay":"HIDDEN","enableBehavior":"ANY","enableWhen":null,"autofillValues":null}],"linkId":"event-group","type":"GROUP","component":null,"prefix":null,"text":"SPDAT","briefText":null,"readonlyText":null,"helperText":null,"required":false,"warnIfEmpty":false,"hidden":false,"readOnly":false,"repeats":false,"mapping":null,"pickListReference":null,"size":null,"assessmentDate":null,"prefill":false,"bounds":null,"pickListOptions":null,"initial":null,"dataCollectedAbout":null,"disabledDisplay":"HIDDEN","enableBehavior":"ANY","enableWhen":null,"autofillValues":null}]}',
```

Comment:
For the sake of https://github.com/open-path/Green-River/issues/5916, lets make these `linkId`s compliant with the  constraints we plan to add. So no periods, no hyphens, and no starting with numbers.